src/main/java/com/intuit/graphql/orchestrator/batch/GraphQLServiceBatchLoader.java

```diff
@@ -114,10 +114,9 @@ private BatchFieldAuthorization getDefaultOrCustomFieldAuthorization(GraphQLCont
       MergedField filteredRootField = result.getMergedField();
       if (filteredRootField != null) {
         filteredRootField.getFields().stream()
-            .map(field -> serviceMetadata.hasFieldResolverDirective()
-                ? removeFieldsWithExternalTypes(field, getRootFieldDefinition(key.getExecutionStepInfo()).getType())
-                : field
-            )
+            .map(field -> (serviceMetadata.hasFieldResolverDirective() || serviceMetadata.isFederationService())
```

Comment:
to make this shorter, we can have

`serviceMetadata.shouldRemoveExternalFields()`